pkg/gatewayserver/io/io.go

```diff
@@ -652,12 +655,15 @@ func (c *Connection) ScheduleDown(path *ttnpb.DownlinkPath, msg *ttnpb.DownlinkM
 			}
 			f = c.scheduler.ScheduleAt
 			settings.Time = request.AbsoluteTime
+			maxDelay = time.Until(request.AbsoluteTime.AsTime())
 		case ttnpb.Class_CLASS_C:
 			if request.AbsoluteTime != nil {
 				f = c.scheduler.ScheduleAt
 				settings.Time = request.AbsoluteTime
+				maxDelay = time.Until(request.AbsoluteTime.AsTime())
 			} else {
 				f = c.scheduler.ScheduleAnytime
+				maxDelay = scheduling.DutyCycleWindow + 5*time.Second
```

Comment:
Lots of "anytime" schedules can fill up multiple duty-cycle windows.

That NS doesn't schedule these is great, but class C downlink can also be scheduled via roaming.

Reply:
Are you sure that you can schedule class C downlinks via roaming ? There is no support for that in either Packet Broker, nor BI to the best of knowledge - you cannot schedule something without an uplink token, and these do not exist in class C.

Regarding filling multiple duty cycle windows - I understand the argument that this will limit the delay to 'only' one hour, but how else could we track information about this issue ? We have no reproduction steps, and we only see effects. At least with this form of primitive logging we can try to settle a set of affected gateways and limit the damage that the invalid delay could do.

Reply:
My point is that this is not just logging but actually setting `delay` and potentially changing behavior.

So let's make this PR focus on logging only, and then have another PR for letting GS return errors when the scheduling would happen too far in the future.